src/components/pdf-upload/PdfUpload.tsx

```diff
@@ -0,0 +1,190 @@
+        aria-label="Upload PDF file"
+        onClick={() => fileInputRef.current?.click()}
+        onKeyDown={(e) => {
+          if (e.key === "Enter" || e.key === " ") {
+            fileInputRef.current?.click();
+          }
+        }}
+        onKeyUp={(e) => {
+          if (e.key === "Enter" || e.key === " ") {
+            fileInputRef.current?.click();
+          }
+        }}
+        onDrop={handleDrop}
+        onDragOver={handleDragOver}
+      >
+        <input
+          ref={fileInputRef}
+          type="file"
+          accept="application/pdf"
+          className="hidden"
+          id="pdf-upload-input"
+          onChange={handleFileSelect}
+        />
+
+        {selectedFile ? (
+          <div className="space-y-2">
+            <div className="text-ob-primary font-medium">
+              ✓ {selectedFile.name}
+            </div>
+            <div className="text-ob-base-200 text-sm">
+              {(selectedFile.size / 1024 / 1024).toFixed(2)} MB
+            </div>
+          </div>
+        ) : (
+          <div className="space-y-2">
+            <div className="text-ob-base-200">
+              <svg
+                className="mx-auto h-12 w-12"
+                stroke="currentColor"
+                fill="none"
+                viewBox="0 0 48 48"
+                aria-hidden="true"
+              >
+                <path
+                  d="M28 8H12a4 4 0 00-4 4v20m32-12v8m0 0v8a4 4 0 01-4 4H12a4 4 0 01-4-4v-4m32-4l-3.172-3.172a4 4 0 00-5.656 0L28 28M8 32l9.172-9.172a4 4 0 015.656 0L28 28m0 0l4 4m4-24h8m-4-4v8m-12 4h.02"
+                  strokeWidth={2}
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                />
+              </svg>
+            </div>
+            <div className="text-ob-base-300 font-medium">
+              Click to select or drag and drop
+            </div>
+            <div className="text-ob-base-200 text-sm">PDF files only</div>
+          </div>
+        )}
+      </button>
+
+      {!isValid && (
+        <div className="text-ob-destructive text-sm">
+          Please select a valid PDF file
+        </div>
+      )}
+
+      {/* Description Input */}
+      <div className="space-y-2">
+        <label
+          htmlFor="pdf-description"
+          className="text-ob-base-300 text-sm font-medium"
+        >
+          Description (optional)
+        </label>
+        <Input
+          id="pdf-description"
+          placeholder="Enter a description for this PDF..."
+          value={description}
+          onValueChange={(value) => setDescription(value)}
+          disabled={loading}
+        />
+      </div>
+
+      {/* Tags Input */}
+      <div className="space-y-2">
+        <label
+          htmlFor="pdf-tags"
+          className="text-ob-base-300 text-sm font-medium"
+        >
+          Tags (optional)
+        </label>
+        <Input
+          id="pdf-tags"
+          placeholder="Enter tags separated by commas..."
```

Comment:
sg (tho let me see how it feels when i try it) - i think we should do a feature request to enter a new tag on enter and have pills for each

Reply:
sounds great. added an issue for it.